src/urllib3/exceptions.py

```diff
@@ -258,3 +258,11 @@ def __init__(self, defects, unparsed_data):
 class UnrewindableBodyError(HTTPError):
     "urllib3 encountered an error when trying to rewind a body"
     pass
+
+
+class NameResolutionError(HTTPError):
```

Comment:
Could we multiclass this error as `HTTPError, socket.gaierror` so that code paths catching DNS errors aren't broken?